Add uaelib trap 41 (WarpMode) to toggle emulator warp mode from the guest.

emulib_WarpMode(mode) calls amiberry's warpmode(), letting Amiga code turn
turbo/warp emulation on and off.
Index: demoscene-toolchain/submodules/amiberry/src/uaelib.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/amiberry/src/uaelib.cpp
+++ demoscene-toolchain/submodules/amiberry/src/uaelib.cpp
@@ -26,6 +26,7 @@
 #include "gensound.h"
 #include "picasso96.h"
 #include "filesys.h"
+#include "inputdevice.h"
 #ifdef AMIBERRY
 #include "target.h"
 #include "parser.h" // for midi ports
@@ -766,6 +767,13 @@ static uae_u32 uaelib_midi(TrapContext *
 
 #include "uaelib_log.cpp"
 
+/* Toggle the emulator warp (turbo) mode from guest code. */
+static uae_u32 emulib_WarpMode(uae_u32 mode)
+{
+  warpmode(mode ? 1 : 0);
+  return 1;
+}
+
 static uae_u32 uaelib_demux_common(TrapContext *ctx, uae_u32 ARG0, uae_u32 ARG1, uae_u32 ARG2, uae_u32 ARG3, uae_u32 ARG4, uae_u32 ARG5)
 {
 	write_log("%ld\n",ARG0);
@@ -790,6 +798,7 @@ static uae_u32 uaelib_demux_common(TrapC
 		case 15: return emulib_Debug();
 
 		case 40: return emulib_Log(ctx, ARG1);
+		case 41: return emulib_WarpMode(ARG1);
 
 		case 68: return emulib_Minimize();
 		case 69: return emulib_ExecuteNativeCode();
